ui/app/services/permissions.js

```diff
@@ -0,0 +1,36 @@
+import Service, { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+// decide what happens when no policies are found
+// because requesting the root token's ACL policy always returns nil
+// https://github.com/hashicorp/vault/pull/4386/files#diff-a145099c8e3d917858619dcfb2ae09b9R3464
+
+export default Service.extend({
+  paths: null,
+  store: service(),
+
+  getPaths: task(function*() {
+    if (this.get('paths')) {
+      return;
+    }
+    let resp = yield this.get('store')
+      .adapterFor('permissions')
+      .query();
+    this.setPaths(resp);
+    return;
+  }),
+
+  setPaths(resp) {
+    this.set('paths', resp.data);
+  },
+
+  hasPermission(pathName) {
+    if (this.get('paths')) {
+      const paths = this.get('paths');
+      return Object.keys(paths).some(pathType => {
+        return paths[pathType].hasOwnProperty(pathName);
```

Comment:
This will likely have to change because globbed paths act as a prefix, so it won't be an exact match. We'd also want to check globbed paths first, and then exact paths since longest path wins. How should we handle `""` in globbed paths (that's the value for the path if you have a policy on `"*"`)?